port/zos390/omriarv64.c

```diff
@@ -75,25 +75,48 @@ void * omrallocate_1M_fixed_pages(int *numMBSegments, int *userExtendedPrivateAr
 	segments = *numMBSegments;
 	wgetstor = lgetstor;
 
-	switch (useMemoryType) {
-	case ZOS64_VMEM_ABOVE_BAR_GENERAL:
-		__asm(" IARV64 REQUEST=GETSTOR,COND=YES,SADMP=NO,CONTROL=UNAUTH,PAGEFRAMESIZE=1MEG,"\
-				"SEGMENTS=(%2),ORIGIN=(%1),TTOKEN=(%4),RETCODE=%0,MF=(E,(%3))"\
-				::"m"(iarv64_rc),"r"(&origin),"r"(&segments),"r"(&wgetstor),"r"(ttkn));
-		break;
-	case ZOS64_VMEM_2_TO_32G:
-		__asm(" IARV64 REQUEST=GETSTOR,COND=YES,SADMP=NO,USE2GTO32G=YES,"\
-				"CONTROL=UNAUTH,PAGEFRAMESIZE=1MEG,"\
-				"SEGMENTS=(%2),ORIGIN=(%1),TTOKEN=(%4),RETCODE=%0,MF=(E,(%3))"\
-				::"m"(iarv64_rc),"r"(&origin),"r"(&segments),"r"(&wgetstor),"r"(ttkn));
-		break;
-	case ZOS64_VMEM_2_TO_64G:
-		__asm(" IARV64 REQUEST=GETSTOR,COND=YES,SADMP=NO,USE2GTO64G=YES,"\
-				"CONTROL=UNAUTH,PAGEFRAMESIZE=1MEG,"\
-				"SEGMENTS=(%2),ORIGIN=(%1),TTOKEN=(%4),RETCODE=%0,MF=(E,(%3))"\
-				::"m"(iarv64_rc),"r"(&origin),"r"(&segments),"r"(&wgetstor),"r"(ttkn));
-		break;
-	}
+    if (startAddress == (void*)0) {
+        switch (useMemoryType) {
+        case ZOS64_VMEM_ABOVE_BAR_GENERAL:
+            __asm(" IARV64 REQUEST=GETSTOR,COND=YES,SADMP=NO,CONTROL=UNAUTH,PAGEFRAMESIZE=1MEG,"\
+                    "SEGMENTS=(%2),ORIGIN=(%1),TTOKEN=(%4),RETCODE=%0,MF=(E,(%3))"\
+                    ::"m"(iarv64_rc),"r"(&origin),"r"(&segments),"r"(&wgetstor),"r"(ttkn));
```

Comment:
Once compiled would this IARV64 call work on old ZOS versions without support for `ORIGIN`? I guess answer is "yes" just want to double check.

Reply:
`ORIGIN` has been used in the existing `IARV64` call and it's a `required` output parameter since zos v2r1. So there's no reason why it wouldn't work.
https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.1.0/com.ibm.zos.v2r1.ieaa900/gstor.htm

Reply:
sorry, I thought about `INORIGIN`and I put in wrong place. However is `startAddress ` is not NULL and so you add `INORIGIN` would it work on old ZOS system does not support hints?

Reply:
If not you might need to set startAddress to NULL at caller side if allocation hint is not supported

Reply:
I confirmed with David that `INORIGIN` would be ignored by old zOS.